spec/support/requests_spec_helper.rb

```diff
@@ -1,9 +1,13 @@
+require "support/default_as_json"
+
 module RequestSpecHelper
   RSpec.configure do |config|
     config.before(:example, :type => :request) do
       allow(Rails.application.config.action_dispatch).to receive(:show_exceptions)
       allow(Rails.application.config).to receive(:consider_all_requests_local).and_return(false)
     end
+
+    config.include DefaultAsJson, :type => :request
```

Comment:
Out of curiosity...is `.prepend` available here (and does it then call `.included`?  If so, then you may not need the extra module nesting in default_as_json.rb.

Reply:
@Fryguy I followed https://relishapp.com/rspec/rspec-core/docs/helper-methods/define-helper-methods-in-a-module which mentioned `.include` and `.extend`, but it *does* look like prepend exists https://www.rubydoc.info/github/rspec/rspec-core/RSpec%2FCore%2FConfiguration:prepend.

Reply:
@eclarizio Is there a way to override the format of data, with this change we would default to :json  how would one of the specs override it to use non json data.

Reply:
@mkanoor Yes, just need to do `let(:default_as) { :text }` or whatever format you want.